File: leetcode-js/074-search-matrix.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    let row = matrix.length - 1;
    let col = 0;

    while (row >= 0 && col < matrix[0].length) {
        if (matrix[row][col] > target) {
            row -= 1;
        } else if (matrix[row][col] < target) {
            col += 1;
        } else {
            return true;
        }
    }

    return false;
};

File: leetcode-java/074-search-matrix.java
public class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n = matrix.length;
        int m = matrix[0].length;
        int l = 0;
        int r = m * n - 1;
        while (l != r) {
            int mid = (l + r - 1) / 2;
            if (matrix[mid / m][mid % m] < target) {
                l = mid + 1;
            } else {
                r = mid;
            }
        }
        return matrix[r / m][r % m] == target;
    }
}
